src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -525,6 +605,18 @@ struct NecroticShard : public ScriptedAI
         {
             HandleShardMinionSpawnerSmall();
         });
+        AddCustomAction(11, 15000u, [&]() // Check if there are a summoning circle and a necropolis nearby, otherwise despawn
+        {
+            if (!m_creature)
+                return;
+            if (!GetClosestGameObjectWithEntry(m_creature, GOBJ_SUMMON_CIRCLE, 2.f) || !GetClosestCreatureWithEntry(m_creature, NPC_NECROPOLIS_HEALTH, 1000.f))
```

Comment:
Again need to find a suitable range instead of 1000.f
This code is here because of how "Battles won" work on server restart. The invasion resumes, and if one necropolis was defeated prior to restart but the others weren't, it means that now 1 or 2 necropoli, depending on the zone, will respawn in a random necropolis location. GameObjects like the summoning circle or creatures like the Necrotic Shard however may either still be despawned from that previous victory or remain spawned at a place that has no necropolis. This checks for this after 15 seconds to leave enough time to load all gameobjects nearby and then every 60 seconds. If no player is in a zone, gameobject load status is kinda undefined, so this is there to make sure that variance is managed

Reply:
You do not need to fret about it that much. If server restarts, thats on the person restarting them. We just need to reload in a roughly equal state without forgetting wins. Server should generally not restart more than once a week with these events active, and the saving is mainly there to avoid detrimental effects of crashes.

Reply:
It's true that server restarts shouldn't be a major point of concern, but I don't really feel comfortable restarting a server into an undefined state (well even more undefined than usually) particularly because some of these states can leave the attack unwinnable, requiring manual intervention from GMs or above. Of course, that's their problem, but not handling this cleanly also reflects on us imo